File: demo-app/app/src/javaApp/java/com/launchkey/android/authenticator/demo/ui/fragment/CustomUnlinkFragment2.java
package com.launchkey.android.authenticator.demo.ui.fragment;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.launchkey.android.authenticator.demo.R;
import com.launchkey.android.authenticator.demo.util.Utils;
import com.launchkey.android.authenticator.sdk.AuthenticatorManager;
import com.launchkey.android.authenticator.sdk.SimpleOperationCallback;
import com.launchkey.android.authenticator.sdk.error.BaseError;

/**
 * Created by armando on 7/20/16.
 */
public class CustomUnlinkFragment2 extends BaseDemoFragment {

    private ProgressDialog mUnlinkingDialog;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        mUnlinkingDialog = new ProgressDialog(getActivity(), R.style.Theme_WhiteLabel_Dialog);
        mUnlinkingDialog.setIndeterminate(true);
        mUnlinkingDialog.setCancelable(false);
        mUnlinkingDialog.setMessage("Custom Unlinking...");

        unlink();
    }

    private void unlink() {

        Utils.show(mUnlinkingDialog);

        AuthenticatorManager.getInstance().unlinkCurrentDevice(new SimpleOperationCallback() {

            @Override
            public void onResult(boolean successful, BaseError error, Object o) {

                Utils.dismiss(mUnlinkingDialog);

                if (successful) {
                    if (getActivity() != null && !getActivity().isFinishing()) {
                        getActivity().finish();
                    }
                } else {

                    String message = Utils.getMessageForBaseError(error);
                    Utils.alert(getActivity(), "Error", message);
                }
            }
        });
    }
}

File: demo-app/app/src/kotlinApp/java/com/launchkey/android/authenticator/demo/ui/fragment/CustomUnlinkFragment2.kt
package com.launchkey.android.authenticator.demo.ui.fragment

import android.app.ProgressDialog
import android.os.Bundle

import com.launchkey.android.authenticator.demo.R
import com.launchkey.android.authenticator.demo.util.Utils
import com.launchkey.android.authenticator.sdk.AuthenticatorManager
import com.launchkey.android.authenticator.sdk.SimpleOperationCallback
import com.launchkey.android.authenticator.sdk.error.BaseError

/**
 * Created by armando on 7/20/16.
 */
class CustomUnlinkFragment2 : BaseDemoFragment() {

    private var mUnlinkingDialog: ProgressDialog? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        mUnlinkingDialog = ProgressDialog(activity, R.style.Theme_WhiteLabel_Dialog)
        mUnlinkingDialog!!.isIndeterminate = true
        mUnlinkingDialog!!.setCancelable(false)
        mUnlinkingDialog!!.setMessage("Custom Unlinking...")

        unlink()
    }

    private fun unlink() {

        Utils.show(mUnlinkingDialog)

        AuthenticatorManager.getInstance().unlinkCurrentDevice(object : SimpleOperationCallback<Any> {

            override fun onResult(successful: Boolean, error: BaseError?, o: Any?) {

                Utils.dismiss(mUnlinkingDialog)

                if (successful) {
                    if (activity != null && !activity!!.isFinishing) {
                        activity!!.finish()
                    }
                } else {

                    val message = Utils.getMessageForBaseError(error)
                    Utils.alert(activity, "Error", message)
                }
            }
        })
    }
}
